src/pke/lib/schemerns/rns-pke.cpp

```diff
@@ -208,8 +209,17 @@ std::shared_ptr<std::vector<DCRTPoly>> PKERNS::EncryptZeroCore(
                    ? DCRTPoly(dgg, elementParams, Format::EVALUATION)
                    : DCRTPoly(tug, elementParams, Format::EVALUATION);
 
-  DCRTPoly e0(dgg, elementParams, Format::EVALUATION);
-  DCRTPoly e1(dgg, elementParams, Format::EVALUATION);
+  DCRTPoly e0;
+  DCRTPoly e1;
+  
+  if (dgg.GetStd() == 1) {
```

Comment:
as we have discussed, you should have created a function checking the state of dgg. dgg.GetStd() is wrong

Reply:
As discussed, you were going to add a IsValid function to dgg because it is applicable outside of PRE too. I can add it if you want me to instead.